sky/backends/backend_utils.py

```diff
@@ -726,13 +730,18 @@ def write_cluster_config(to_provision: 'resources.Resources',
                 'ssh_private_key': (None if auth_config is None else
                                     auth_config['ssh_private_key']),
             }),
-        dryrun=dryrun,
     )
     config_dict['cluster_name'] = cluster_name
     config_dict['ray'] = yaml_path
     if dryrun:
         return config_dict
     _add_auth_to_cluster_config(cloud, yaml_path)
+    # Delay the optimization of the config until the authentication files is added.
+    if not isinstance(cloud, clouds.Local):
+        # Only optimize the file mounts for public clouds now, as it has not been
+        # fully tested local yet.
```

Comment:
nit: "as local has not been fully tested yet"

Reply:
Oops, good catch. Fixed. Thanks!